schema/2.0.0/corpora_schema.md

```diff
@@ -0,0 +1,257 @@
+    Summarizing, any cell barcodes that are removed from the data MUST be filtered from both raw and final matrices and genes MUST NOT be filtered from the raw matrix.
+*   Any genes that publishers wish to filter from the final matrix MAY have their values replaced by a language appropriate "null" value (e.g. [`np.nan`](https://numpy.org/doc/stable/reference/constants.html#numpy.nan) for python), which will mask them from exploration.
+*   Additional layers provided at author discretion MAY be stored using author-selected keys, but MUST have the same cells and genes as other layers.
+
+The following table describes the matrix data and layers requirements that are **assay-specific**. If an entry in the table is empty, the cellxgene schema does not have any other requirements on data in those layers beyond the ones listed above.
+
+| Assay | "raw" required? | "raw" requirements | "final" required? | "final" requirements | Other layers |
+|-|-|-|-|-|-|
+| scRNA-seq (UMI, e.g. 10x v3) | REQUIRED in `AnnData.raw.X` unless no "final" is provided, then `AnnData.X` | Values MUST be de-duplicated molecule counts. | STRONGLY RECOMMENDED |  | OPTIONAL |
+| scRNA-seq (non-UMI, e.g. SS2) | REQUIRED in `AnnData.raw.X` unless no "final" is provided, then `AnnData.X` | Values MUST be one of read counts (e.g. FeatureCounts) or  estimated fragments (e.g. output of RSEM). | STRONGLY RECOMMENDED |  | OPTIONAL |
+| Accessibility (e.g. ATAC-seq, mC-seq) | NOT REQUIRED |  | REQUIRED in `AnnData.X` | Values MUST correspond to ENSEMBL gene identifiers  | OPTIONAL |
+||||
+
+## Integration Metadata
+
+cellxgene requires ontology terms to enable search, comparison, and integration of data.
+Ontology terms for cell metadata MUST use [OBO-format identifiers](http://www.obofoundry.org/id-policy.html), meaning a CURIE (prefixed identifier) of the form **Ontology:Identifier**.
+For example, [EFO:0000001](https://www.ebi.ac.uk/ols/ontologies/efo/terms?short_form=EFO_0000001) is a term in the `EFO` ontology.
+
+When no appropriate ontology value is available, then the most precise accurate term MUST be used.
+For example if the `cell_type` field describes a relay interneuron, but the most specific available term in the CL ontology is [CL:0000099](https://www.ebi.ac.uk/ols/ontologies/cl/terms?obo_id=CL:0000099) for "interneuron", then the interneuron term can be used to fulfill this requirement and ensures that users searching for "neuron" are able to find these data.
+Users will still be able to access more specific cell type annotations that have been submitted with the data (but aren't required by the schema).
+A dataset comprising cells of the human embryo provides a more extreme example.
+In this case, the most precise accurate term is [CL:0000003](https://www.ebi.ac.uk/ols/ontologies/cl/terms?obo_id=CL:0000003)
+for "native cell".
+
+### Required Ontologies
+
+The following ontology dependencies are *pinned* for this version of the schema.
```

Comment:
What are the implications of the pinned versions on the submitter?

Reply:
These are versions of ontologies and gene annotations referenced by cellxgene-schema to validate the related metadata. It implies that when remapping author metadata, the submitter should ensure that they're using identifiers that are appropriate for these versions. Otherwise, validation will fail.

Reply:
So perhaps guidance is need for these cases...
Incredibly likely to occur - Researcher mapped to v19 annotation, and that contains Gene IDs that are no longer present in 2020-A v32. What are they expected to do with those Genes that are not in the pinned version?
Less likely, but possible - Researcher identifies new cell type, or a gap in CL, and requests new term. They get the CL term before the pinned ontology version updates to include that CL term.

Reply:
I defer to @ambrosejcarr - we're aware of the issues. Obsolete terms - my sense is that the curator must use the related web sites to find out what happened to that obsoleted term and re-map it. Freshly baked terms - Dunno. Map to the most accurate existing CL term and include the approved term in parentheses as a suffix? 

We haven't deeply reached how frequently each ontology is updated to consider whether/when we need to update to 2.0.1 to include an updated ontology like CL.

There's also the issue with using the related OLS/ENSEMBL REST API(s) for validation and having versions "drift" in the existing data corpus.

Reply:
> Researcher mapped to v19 annotation, and that contains Gene IDs that are no longer present in 2020-A v32

@jahilton why are IDs typically deleted? Would it be wrong simply to drop them?

> Freshly baked terms - Dunno. Map to the most accurate existing CL term and include the approved term in parentheses as a suffix?

This seems reasonable to me.

Reply:
>Map to the most accurate existing CL term and include the approved term in parentheses as a suffix?

I like this. Possible to validate against 'live' CL version & very handy to update it once the pinned ontology catches up
>@jahilton why are IDs typically deleted? Would it be wrong simply to drop them?

In that scenario, I think the main cause for dropping wouldn't be that a gene was retired/deleted between the run version & the pinned version (those feel OK to drop, maybe even responsible). It would be that the gene changed biotype (was a 'protein coding gene' in v19, but is 'pseudogene' in v32) or the researcher did not filter the same biotypes as 10x does. The latter is the situation where you definitely do not want to drop because the researcher made a decision to include them in their reference. Perhaps using the v32 reference from GENCODE that does not filter biotypes as the pinned reference is a way to adjust for those.

Reply:
RE the special case for new CL terms that do not exist in the pinned version - I've thought about this more and am concerned that we would be asked for similar exceptions for every ontology. It may be preferable to update the schema (2.0.1) on a regular cadence when important changes are noted or discovered in ontologies. @ambrosejcarr and @pablo-gar - thoughts?

Reply:
I agree, unfortunately it's one of the trade offs of having a pinned ontology, but the benefits seem to be greater. And yes this risk can be reduced by schema updates on a frequent basis or per-request basis.